aten/src/ATen/native/cpu/RangeFactoriesKernel.cpp

```diff
@@ -16,30 +16,34 @@ static void linspace_kernel(TensorIterator& iter, Scalar scalar_start, Scalar sc
   AT_DISPATCH_ALL_TYPES_AND_C10_COMPLEX(iter.dtype(), "linspace_cpu", [&]() {
     // step should be of double type for all integral types
     using step_t = std::conditional_t<std::is_integral<scalar_t>::value, double, scalar_t>;
-    int64_t idx(0);
-    scalar_t start = scalar_start.to<scalar_t>();
-    scalar_t end = scalar_end.to<scalar_t>();
-    step_t step = static_cast<step_t>(end - start) / static_cast<step_t>(steps - 1);
+    const scalar_t start = scalar_start.to<scalar_t>();
+    const scalar_t end = scalar_end.to<scalar_t>();
+    const step_t step = static_cast<step_t>(end - start) / (steps - 1);
     int64_t halfway = steps / 2;
-    cpu_serial_kernel_vec(
-        iter,
-        [start, end, step, halfway, steps, &idx]() -> scalar_t {
-          if (idx < halfway) {
-            return start + step * static_cast<step_t>(idx++);
-          } else {
-            return end - step * static_cast<step_t>(steps - (idx++) - 1);
-          }
-        },
-        [start, end, step, halfway, steps, &idx]() -> Vec256<scalar_t> {
-          Vec256<scalar_t> res;
-          if (idx < halfway) {
-            res = Vec256<scalar_t>::arange(start + step * static_cast<step_t>(idx), step);
-          } else {
-            res = Vec256<scalar_t>::arange(end - step * static_cast<step_t>(steps - idx - 1), step);
-          }
-          idx += Vec256<scalar_t>::size();
-          return res;
-        });
+    at::parallel_for(0, steps, internal::GRAIN_SIZE, [&](int64_t p_begin, int64_t p_end) {
+      int64_t idx(p_begin);
+      TensorIterator it(iter);
+      cpu_serial_kernel_vec(
+          it,
+          [start, end, step, halfway, steps, &idx]() -> scalar_t {
+            if (idx < halfway) {
+              return start + step * (idx++);
```

Comment:
Please apply this one
```suggestion
              return start + step * static_cast<step_t>(idx++);
```